src/components/common/HocWithLoading/HocWithLoading.tsx

```diff
@@ -0,0 +1,22 @@
+import clsx from 'clsx';
+import React from 'react';
+
+import { type HocWithLoadingProps } from './types.ts';
+
+const hocWithLoading = <P extends object>(
```

Comment:
A good old HOC! I'm just wondering if having a "composable" component wouldn't be a more flexible solution? 

```tsx
<LoadingSkeleton isLoading={...}>
  <Component />
</LoadingSkeleton>
```

At the same time I don't have a strong argument against a HOC, so feel free to leave it in

Reply:
Thanks for reviewing this @jakubcolony! It's a good thing you mentioned this because it reminded me of some stashed changes I have about passing KYC state! 😂 

I thought this a HOC was really neat because it saves us a few lines of code if used extensively and all we have to worry about really is making it load or not, and some dimensions config as well. But having said that, I think composition would be preferable as it aligns with the composition changes I did for the KYC page.

I'll be making this change and will also push the stashed changes I was referring to. It's basically just wrapping the page in a Context Provider which also exposes the mutation we use to get the KYC. Should be helpful when we want to refetch the data at any given time i.e. when the KYC flow has succeeded, we can refetch to refresh the data of the page 👌